src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,30 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws RegistrationException {
+        if (user.getLogin() == null || user.getLogin().length() < 6) {
+            throw new RegistrationException("Login should be at least 6 characters");
+        }
+        if (user.getPassword() == null || user.getPassword().length() < 6) {
+            throw new RegistrationException("Password should be at least 6 characters");
+        }
+        if (user.getAge() == null || user.getAge() < 18) {
+            throw new RegistrationException("Age should be at least 18 years old");
+        }
+        for (User current : Storage.people) {
+            if (current.getLogin().equals(user.getLogin())) {
+                throw new RegistrationException("User with such login already exist");
+            }
+        }
+        Storage.people.add(user);
```

Comment:
Use low-level methods like `Storage.people.add()` directly instead of `RegistrationService.register()` method to add users for testing. Because we can not use the same method which is under a test to insert some mock test data into our storage since we can not guarantee its correctness yet. However, in this implementation, the direct use of `Storage.people.add()` is correct since it's the actual operation to add a user. Just ensure the tests are not using `register` method to prepare test data.